### ISSUE TITLE
pytest-vcr==1.0.0 fails with pytest==4.0.0 if one specifies --vcr-record or --vcr-record-mode

### ISSUE THREAD

**Issue by immerrr** [ktosiek/pytest-vcr#16]
There are two things in play here:
- pytest-vcr tries to report `vcr-record-mode` parameter that is deprecated in favour of `vcr-record` as a warning
- the way pytest-vcr reports it is via a functionality that is deprecated and removed in pytest 4

As a result, all tests marked with `pytest.vcr` marker fail during collection with the following message:
```
$ pytest --vcr-record-mode=none
<...snip...>
2018-11-15T17:59:15.861968589Z     @pytest.fixture(scope='module')
2018-11-15T17:59:15.862173278Z     def vcr(request, pytestconfig, vcr_config, vcr_cassette_dir, ):
2018-11-15T17:59:15.862335576Z         """The VCR instance"""
2018-11-15T17:59:15.862537760Z         if request.config.getoption('--vcr-record-mode'):
2018-11-15T17:59:15.862692441Z             pytestconfig.warn("C1",
2018-11-15T17:59:15.862932122Z >                             "--vcr-record-mode has been deprecated and will be removed in a future "
2018-11-15T17:59:15.863159585Z                               "release. Use --vcr-record instead.")
2018-11-15T17:59:15.863405921Z E           _pytest.warning_types.RemovedInPytest4Warning: config.warn has been deprecated, use warnings.warn instead
2018-11-15T17:59:15.863569363Z 
2018-11-15T17:59:15.863729891Z /usr/local/lib/python3.6/site-packages/pytest_vcr.py:69: RemovedInPytest4Warning
<...snip...>
```

If you have pipenv installed, you can reproduce it as follows:
```
$ mkdir test-pytest-vcr &&  cd test-pytest-vcr
$ pipenv install pytest==4.0.0 pytest-vcr==1.0.0 > /dev/null
  🐍   ▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉▉ 17/17 — 00:00:01
$ cat <<EOF > test_foo.py
import pytest
@pytest.mark.vcr
def test_foo():
    pass

EOF
$ pipenv run pytest --vcr-record-mode=none
==================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.12, pytest-4.0.0, py-1.7.0, pluggy-0.8.0
rootdir: /home/immerrr/src/test-pytest-vcr, inifile:
plugins: vcr-1.0.0
collected 1 item                                                                                                                                                                             

test_foo.py E                                                                                                                                                                          [100%]

=========================================================================================== ERRORS ===========================================================================================
_________________________________________________________________________________ ERROR at setup of test_foo _________________________________________________________________________________

request = <SubRequest '_vcr_marker' for <Function 'test_foo'>>

    @pytest.fixture(autouse=True)
    def _vcr_marker(request):
        marker = request.node.get_closest_marker('vcr')
        if marker:
>           request.getfixturevalue('vcr_cassette')

/mnt/extraspace/virtualenvs/test-pytest-vcr-ni0iqlSt/lib/python2.7/site-packages/pytest_vcr.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

request = <SubRequest 'vcr' for <Function 'test_foo'>>, pytestconfig = <_pytest.config.Config object at 0x7f5e57bc8ed0>, vcr_config = {}
vcr_cassette_dir = '/home/immerrr/src/test-pytest-vcr/cassettes'

    @pytest.fixture(scope='module')
    def vcr(request, pytestconfig, vcr_config, vcr_cassette_dir, ):
        """The VCR instance"""
        if request.config.getoption('--vcr-record-mode'):
            pytestconfig.warn("C1",
>                             "--vcr-record-mode has been deprecated and will be removed in a future "
                              "release. Use --vcr-record instead.")
E           RemovedInPytest4Warning: config.warn has been deprecated, use warnings.warn instead

/mnt/extraspace/virtualenvs/test-pytest-vcr-ni0iqlSt/lib/python2.7/site-packages/pytest_vcr.py:69: RemovedInPytest4Warning
================================================================================== 1 error in 0.04 seconds ===================================================================================
```

**Comment by immerrr** [comment]
Same message *was* shown for the newer `--vcr-record` parameter:
```
$ pipenv run pytest --vcr-record=none
==================================================================================== test session starts =====================================================================================
platform linux2 -- Python 2.7.12, pytest-4.0.0, py-1.7.0, pluggy-0.8.0
rootdir: /home/immerrr/src/test-pytest-vcr, inifile:
plugins: vcr-1.0.0
collected 1 item                                                                                                                                                                             

test_foo.py E                                                                                                                                                                          [100%]

=========================================================================================== ERRORS ===========================================================================================
_________________________________________________________________________________ ERROR at setup of test_foo _________________________________________________________________________________

request = <SubRequest '_vcr_marker' for <Function 'test_foo'>>

    @pytest.fixture(autouse=True)
    def _vcr_marker(request):
        marker = request.node.get_closest_marker('vcr')
        if marker:
>           request.getfixturevalue('vcr_cassette')

/mnt/extraspace/virtualenvs/test-pytest-vcr-ni0iqlSt/lib/python2.7/site-packages/pytest_vcr.py:45: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

request = <SubRequest 'vcr' for <Function 'test_foo'>>, pytestconfig = <_pytest.config.Config object at 0x7f21b71c1e90>, vcr_config = {}
vcr_cassette_dir = '/home/immerrr/src/test-pytest-vcr/cassettes'

    @pytest.fixture(scope='module')
    def vcr(request, pytestconfig, vcr_config, vcr_cassette_dir, ):
        """The VCR instance"""
        if request.config.getoption('--vcr-record-mode'):
            pytestconfig.warn("C1",
>                             "--vcr-record-mode has been deprecated and will be removed in a future "
                              "release. Use --vcr-record instead.")
E           RemovedInPytest4Warning: config.warn has been deprecated, use warnings.warn instead

/mnt/extraspace/virtualenvs/test-pytest-vcr-ni0iqlSt/lib/python2.7/site-packages/pytest_vcr.py:69: RemovedInPytest4Warning
================================================================================== 1 error in 0.04 seconds ===================================================================================
``` 

But it should be fixed with #14

**Comment by ktosiek** [comment]
I think I understand the issue now. It should be fixed in 1.0.1, so I'm closing this issue, but please comment if it still happens there.
